### PR TITLE
Feature/cache invalidation

### PR DESCRIPTION
**Pull request by rubensworks**
This implements cache invalidation, as required for #341.

@RubenVerborgh With this, you will be able to call `invalidateHttpCache()` (with an optional `pageUrl` arg) on a Comunica engine, which is needed to handle RubenVerborgh/LDflex-Comunica#10.

### CODE REVIEW

**Comment by RubenVerborgh** [comment]
Thanks for this!

File: packages/actor-init-sparql/lib/ActorInitSparql-browser.ts [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 import {IActionOptimizeQueryOperation,
   IActorOptimizeQueryOperationOutput} from "@comunica/bus-optimize-query-operation";
 import {Bindings, IActionQueryOperation, IActorQueryOperationOutput} from "@comunica/bus-query-operation";
 import {KEY_CONTEXT_SOURCES} from "@comunica/bus-rdf-resolve-quad-pattern";
 import {IActionSparqlParse, IActorSparqlParseOutput} from "@comunica/bus-sparql-parse";
-import {IActionSparqlSerialize} from "@comunica/bus-sparql-serialize";
 import {IActionRootSparqlParse, IActorOutputRootSparqlParse,
   IActorTestRootSparqlParse} from "@comunica/bus-sparql-serialize";
 import {IActorSparqlSerializeOutput} from "@comunica/bus-sparql-serialize";
+import {IActionSparqlSerialize} from "@comunica/bus-sparql-serialize";
```
Comment (joachimvh) [inline packages/actor-init-sparql/lib/ActorInitSparql-browser.ts]: minor, but these imports could be combined

File: packages/actor-init-sparql/lib/ActorInitSparql-browser.ts [hunk 2]
```diff
@@ -186,6 +189,16 @@ export class ActorInitSparql extends ActorInit implements IActorInitSparqlArgs {
     return (await this.mediatorSparqlSerialize.mediate({ context, handle, handleMediaType: mediaType })).handle;
   }
 
+  /**
+   * Invalidate all internal caches related to the given page URL.
+   * If no page URL is given, then all pages will be invalidated.
+   * @param {string} pageUrl The page URL to invalidate.
+   * @return {Promise<any>} A promise resolving when the caches have been invalidated.
+   */
+  public invalidateHttpCache(pageUrl?: string): Promise<any> {
```
Comment (joachimvh) [inline packages/actor-init-sparql/lib/ActorInitSparql-browser.ts]: Can this function be called on the newEngine object, or where is it used?

File: packages/bus-http-invalidate/lib/ActorHttpInvalidateListenable.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import {IActorArgs, IActorTest} from "@comunica/core";
+import {ActorHttpInvalidate, IActionHttpInvalidate, IActorHttpInvalidateOutput} from "./ActorHttpInvalidate";
+
+/**
+ * An ActorHttpInvalidate actor that allows listeners to be attached.
```
[the remark this answers is not in the document]
Comment (rubensworks) [reply packages/bus-http-invalidate/lib/ActorHttpInvalidateListenable.ts]: I agree. Something for Comunica 2.0 :-)

File: packages/actor-init-sparql/lib/ActorInitSparql-browser.ts (hunk 2 above)
Comment (rubensworks) [reply packages/actor-init-sparql/lib/ActorInitSparql-browser.ts]: Yep! Because the `newEngine()` returns `ActorInitSparql`.

**Pull request merged**